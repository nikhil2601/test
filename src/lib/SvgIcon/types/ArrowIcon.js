import PropTypes from 'prop-types';
import React from 'react';

import SvgIcon from '../SvgIcon';

/**
 * Render a arrow based on a given direction
 *
 * @method renderDirectionalArrowIcon
 * @param  {string}                   direction The direction of the arrow
 * @return {JSX}
 */
const renderDirectionalArrowIcon = direction => {
    switch (direction) {
        case 'up':
            return (
                <path d="M13,20H11V8L5.5,13.5L4.08,12.08L12,4.16L19.92,12.08L18.5,13.5L13,8V20Z" />
            );
        case 'right':
            return (
                <path d="M4,11V13H16L10.5,18.5L11.92,19.92L19.84,12L11.92,4.08L10.5,5.5L16,11H4Z" />
            );
        case 'down':
            return (
                <path d="M11,4H13V16L18.5,10.5L19.92,11.92L12,19.84L4.08,11.92L5.5,10.5L11,16V4Z" />
            );
        case 'left':
            return (
                <path d="M20,11V13H8L13.5,18.5L12.08,19.92L4.16,12L12.08,4.08L13.5,5.5L8,11H20Z" />
            );
        default:
            return null;
    }
};

const ArrowIcon = ({ direction, setRef, ...rest }) => (
    <SvgIcon {...rest} ref={setRef}>
        {renderDirectionalArrowIcon(direction)}
    </SvgIcon>
);

ArrowIcon.propTypes = {
    direction: PropTypes.oneOf(['up', 'right', 'down', 'left']),
    setRef: PropTypes.object,
    viewBox: PropTypes.string,
};

ArrowIcon.defaultProps = {
    direction: 'up',
    setRef: null,
    viewBox: '0 0 24 24',
};

export default ArrowIcon;
