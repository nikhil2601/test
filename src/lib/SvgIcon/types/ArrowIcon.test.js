import React from 'react';
import { shallow } from 'enzyme';

import ArrowIcon from './ArrowIcon';

describe('<ArrowIcon />', () => {
    let wrapper;

    it('should render correctly by default', () => {
        wrapper = shallow(<ArrowIcon />);
        expect(wrapper).toMatchSnapshot();
    });

    it('should render correctly as Up Arrow', () => {
        wrapper = shallow(<ArrowIcon direction="up" />);
        expect(wrapper).toMatchSnapshot();
    });

    it('should render correctly as Down Arrow', () => {
        wrapper = shallow(<ArrowIcon direction="down" />);
        expect(wrapper).toMatchSnapshot();
    });

    it('should render correctly as Left Arrow', () => {
        wrapper = shallow(<ArrowIcon direction="left" />);
        expect(wrapper).toMatchSnapshot();
    });

    it('should render correctly as Right Arrow', () => {
        wrapper = shallow(<ArrowIcon direction="right" />);
        expect(wrapper).toMatchSnapshot();
    });
});
