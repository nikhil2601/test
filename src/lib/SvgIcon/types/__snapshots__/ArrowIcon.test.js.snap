// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ArrowIcon /> should render correctly as Down Arrow 1`] = `
<SvgIcon
  color="inherit"
  height="1rem"
  margin={null}
  opacity={null}
  padding={null}
  titleAccess={null}
  viewBox="0 0 24 24"
  width="1rem"
>
  <path
    d="M11,4H13V16L18.5,10.5L19.92,11.92L12,19.84L4.08,11.92L5.5,10.5L11,16V4Z"
  />
</SvgIcon>
`;

exports[`<ArrowIcon /> should render correctly as Left Arrow 1`] = `
<SvgIcon
  color="inherit"
  height="1rem"
  margin={null}
  opacity={null}
  padding={null}
  titleAccess={null}
  viewBox="0 0 24 24"
  width="1rem"
>
  <path
    d="M20,11V13H8L13.5,18.5L12.08,19.92L4.16,12L12.08,4.08L13.5,5.5L8,11H20Z"
  />
</SvgIcon>
`;

exports[`<ArrowIcon /> should render correctly as Right Arrow 1`] = `
<SvgIcon
  color="inherit"
  height="1rem"
  margin={null}
  opacity={null}
  padding={null}
  titleAccess={null}
  viewBox="0 0 24 24"
  width="1rem"
>
  <path
    d="M4,11V13H16L10.5,18.5L11.92,19.92L19.84,12L11.92,4.08L10.5,5.5L16,11H4Z"
  />
</SvgIcon>
`;

exports[`<ArrowIcon /> should render correctly as Up Arrow 1`] = `
<SvgIcon
  color="inherit"
  height="1rem"
  margin={null}
  opacity={null}
  padding={null}
  titleAccess={null}
  viewBox="0 0 24 24"
  width="1rem"
>
  <path
    d="M13,20H11V8L5.5,13.5L4.08,12.08L12,4.16L19.92,12.08L18.5,13.5L13,8V20Z"
  />
</SvgIcon>
`;

exports[`<ArrowIcon /> should render correctly by default 1`] = `
<SvgIcon
  color="inherit"
  height="1rem"
  margin={null}
  opacity={null}
  padding={null}
  titleAccess={null}
  viewBox="0 0 24 24"
  width="1rem"
>
  <path
    d="M13,20H11V8L5.5,13.5L4.08,12.08L12,4.16L19.92,12.08L18.5,13.5L13,8V20Z"
  />
</SvgIcon>
`;
